lib/include/tree_sitter/api.h

```diff
@@ -551,6 +551,12 @@ TSStateId ts_node_next_parse_state(TSNode self);
  */
 TSNode ts_node_parent(TSNode self);
 
+/**
+ * Get a child of 'from_parent' which is a parent to the given node.
+ * The parent may be indirect.
+ */
+TSNode ts_node_closer_parent(TSNode self, TSNode from_parent);
```

Comment:
This is quite obscure. Aren't the changes `ts_node_parent` enough? If not how would has-ancestor be implemented with this? Would it make more sense to expose something like `tsnode_has_ancestor_type()` instead?

Reply:
Yes, but I couldn't think of anything simpler.

Reply:
At least add an example or something. The fact this can be used to efficiently find a path from a root node is useful in itself.

Reply:
I also think the name could be better; maybe `ts_node_parent_from_node` (not great, but more expressive).

Also, would it be possible to consolidate the function with the original `ts_node_parent`? (I currently don't see a non-breaking way -- except for Neovim's Lua API -- but that doesn't mean much.)